### PR TITLE
fix(#2421): fix carousel sliding in wrong direction

### LINKED ISSUE #2421: Carousel: sliding wrong direction
- components: `Carousel`
- reactstrap version `#9.0.1`
- react version `#17.0.2`
- bootstrap version `#5.1.3`

### What is happening?

Carousel slide in wrong direction ie. only from left to right. 

### What should be happening?

Slides should be sliding the same direction as indicators or control ie. moving from right to left and left to right accordingly.

### Steps to reproduce issue

I created a carousel basing of Reactstrap v.8 code which works fine in  v.8. 
However the difference could be observed also in examples:

https://reactstrap.github.io/?path=/docs/components-carousel--carousel - wrong
https://deploy-preview-2356--reactstrap.netlify.app/components/carousel/ - OK 

Thanks for your help in advance.

### PR DESCRIPTION
**Pull request by rj-david**
- [x] Bug fix 
- [ ] New feature 
- [x] My code follows the code style of this project.
- [ ] Chore 
- [ ] Breaking change 
- [x] There is an open issue which this change addresses
- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [x] My commits follow the [Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My change requires a change to [Typescript typings](./types/lib).
  - [ ] I have updated the typings accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

closes #2421

### CODE REVIEW

**Comment by rj-david** [comment]
@phwebi @davidacevedo can you please help merge this

File: package.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
     "prepublishOnly": "npm run build",
     "create-release": "npm run cover && sh ./scripts/release",
     "lint": "eslint src",
-    "dtslint": "dtslint types --expectOnly"
+    "dtslint": "dtslint types --expectOnly",
+    "prepare": "npm run build"
```
Comment (davidacevedo) [inline package.json:29]: I don't think this change is necessary.

File: src/Carousel.js [hunk 2]
```diff
@@ -285,8 +285,8 @@ Carousel.defaultProps = {
   fade: false,
 };
 
-Carousel.childContextTypes = {
-  direction: PropTypes.string
-};
+// Carousel.childContextTypes = {
+//   direction: PropTypes.string
+// };
```
Comment (davidacevedo) [inline src/Carousel.js:290]: Can you delete this code

File: package.json (hunk 1 above)
Comment (rj-david) [reply package.json]: Applicable only when installed from a fork. Removed

**Comment by illiteratewriter** [comment]
@rj-david can you look into this?

**Comment by rpong** [comment]
@illiteratewriter  checked the failing test, shouldn't that be the new behavior? since the carousel is now sliding in the correct direction for this PR?

https://github.com/reactstrap/reactstrap/runs/7405767828?check_suite_focus=true#step:7:185

```
FAIL src/__tests__/Carousel.spec.js
  ● Carousel › items › transitions › should add the appropriate classes when entering right
    expect(received).toEqual(expected)
    Expected value to equal:
      "carousel-item carousel-item-start carousel-item-next"
    Received:
      "carousel-item carousel-item-end carousel-item-prev"
      63 |
      64 |         wrapper.setProps({ in: true });
    > 65 |         expect(wrapper.update().find('div').prop('className')).toEqual(
         |                                                                ^
      66 |           'carousel-item carousel-item-start carousel-item-next',
      67 |         );
      68 |         jest.runTimersToTime(600);
      at Object.<anonymous> (src/__tests__/Carousel.spec.js:65:64)
Test Suites: 1 failed, 88 passed, 89 total
Tests:       1 failed, 876 passed, 877 total
Snapshots:   0 total
Time:        18.479s
Ran all test suites.
npm ERR! Test failed.  See above for more details.
Error: Process completed with exit code 1.
```

**Comment by illiteratewriter** [comment]
@rpong I think the tests are actually failing because of [this issue](https://github.com/enzymejs/enzyme/issues/2189) in enzyme

**Comment by wipei110710** [comment]
@rj-david 
Is there any reactstrap version this pull request merged?

**Comment by illiteratewriter** [comment]
Closed with #2604

**Pull request closed without merging**